### ISSUE TITLE
proxy_connect: upstream read timed out (peer:40.xxx.xx.xxx:443) while connecting to upstream

### ISSUE THREAD

**Issue by sbadyjy** [chobits/ngx_http_proxy_connect_module#191]
### Ⅰ. Issue Description

When an https request comes in, an access.log and an error.log will be generated, but the request and response are normal. 

### Ⅲ. Describe what you expected to happen

The https request comes in, when the request and response are normal, the errror log will not appear

### Ⅳ. How to reproduce it (as minimally and precisely as possible)

1.access.log
```sh
10.135.8.144 - - [28/Apr/2021:14:07:20 +0800] "CONNECT phtcam.xxxxx.com:443 HTTP/1.1" 200 4784 "-" "Apache-HttpClient/4.5.2 (Java/1.8.0_191)"
10.135.8.144 - - [28/Apr/2021:14:07:20 +0800] "CONNECT phtcam.xxxxx.com:443 HTTP/1.1" 200 4784 "-" "Apache-HttpClient/4.5.2 (Java/1.8.0_191)"
10.135.8.144 - - [28/Apr/2021:14:07:20 +0800] "CONNECT phtcam.xxxxx.com:443 HTTP/1.1" 200 4784 "-" "Apache-HttpClient/4.5.2 (Java/1.8.0_191)"
10.135.8.144 - - [28/Apr/2021:14:07:20 +0800] "CONNECT phtcam.xxxxx.com:443 HTTP/1.1" 200 4784 "-" "Apache-HttpClient/4.5.2 (Java/1.8.0_191)"
10.135.8.144 - - [28/Apr/2021:14:07:20 +0800] "CONNECT phtcam.xxxxx.com:443 HTTP/1.1" 200 4784 "-" "Apache-HttpClient/4.5.2 (Java/1.8.0_191)"
```

2.error.log
```sh
2021/04/28 14:07:20 [error] 68744#0: *10 proxy_connect: upstream read timed out (peer:40.xxx.xx.xxx:443) while connecting to upstream, client: 10.135.8.144, server: , request: "CONNECT phtcam.xxxxx.com:443 HTTP/1.1", host: "phtcam.xxxxx.com"
2021/04/28 14:07:20 [error] 68743#0: *12 proxy_connect: upstream read timed out (peer:40.xxx.xx.xxx:443) while connecting to upstream, client: 10.135.8.144, server: , request: "CONNECT phtcam.xxxxx.com:443 HTTP/1.1", host: "phtcam.xxxxx.com"
2021/04/28 14:07:20 [error] 68743#0: *15 proxy_connect: upstream read timed out (peer:40.xxx.xx.xxx:443) while connecting to upstream, client: 10.135.8.144, server: , request: "CONNECT phtcam.xxxxx.com:443 HTTP/1.1", host: "phtcam.xxxxx.com"
2021/04/28 14:07:20 [error] 68742#0: *18 proxy_connect: upstream read timed out (peer:40.xxx.xx.xxx:443) while connecting to upstream, client: 10.135.8.144, server: , request: "CONNECT phtcam.xxxxx.com:443 HTTP/1.1", host: "phtcam.xxxxx.com"
2021/04/28 14:07:20 [error] 68742#0: *21 proxy_connect: upstream read timed out (peer:40.xxx.xx.xxx:443) while connecting to upstream, client: 10.135.8.144, server: , request: "CONNECT phtcam.xxxxx.com:443 HTTP/1.1", host: "phtcam.xxxxx.com"
```
3.nginx.conf
```sh
worker_processes 10;
events {
worker_connections 10240;
}

http {
include mime.types;
default_type application/octet-stream;
log_format main '$remote_addr - $remote_user [$time_local] "$request" '
'$status $body_bytes_sent "$http_referer" '
'"$http_user_agent" "$http_x_forwarded_for"';
access_log /home/stq/nginx/logs/access.log main;
error_log /home/stq/nginx/logs/error.log;
sendfile on;
#tcp_nopush on;
#tcp_nodelay on;
keepalive_timeout 60s;

server {
listen 8888;
resolver 223.5.5.5;
proxy_connect;
proxy_connect_allow 443 563;
proxy_connect_connect_timeout 10s;
proxy_connect_read_timeout 10s;
proxy_connect_send_timeout 10s;
#access_log logs/host.access.log main;
#proxy_ignore_client_abort on;
client_body_timeout 3600s;

location / {
#proxy_pass https://$host$request_uri;
proxy_pass http://$host;
#proxy_pass $scheme://$http_host$request_uri;
proxy_set_header Host $host;
#proxy_set_header X-Real-IP $remote_addr;
#proxy_set_header X-Real-IP $remote_port;
#proxy_pass https://$host$request_uri;
#proxy_ssl_server_name on;
}
#location / {
# proxy_pass $scheme://$http_host$request_uri;
#}

}

}
```

### Ⅵ. Environment:

1. Tengine/Nginx/OpenResty version (use `sbin/nginx -v`):
2. Which [patch](https://github.com/chobits/ngx_http_proxy_connect_module#select-patch) do you use?
nginx 1.20   ngx_http_proxy_connect_module 1018

**Comment by malza321** [comment]
same issue in 1.19.1... and logging is too slow.

**Comment by chobits** [comment]
Link to #175.

Similar issue.

I have understood the problem, but I need to consider how to set right timer and some new instructions or variables. Here, the timer setting have some works and needs more testing.

**Comment by wskyhawk** [comment]
I have the same probalem like this,but I don't know how to solve it.The version of my nginx is "1.20.1".“time_out” is set to 120s.

**Comment by huangyuewu** [comment]
@chobits 我目前也遇到这个问题了，现在还没有修复吗？

**Comment by wskyhawk** [comment]
你好！你的邮件已经收到，我会尽快给你回复，谢谢！

**Comment by chobits** [comment]
hi @huangyuewu huangyuewu 

> @chobits 我目前也遇到这个问题了，现在还没有修复吗？

It has been fixed in `for_pull_request` branch.
check it out in https://github.com/chobits/ngx_http_proxy_connect_module/tree/for_pull_request

已经在 `for_pull_request` branch分支中修复

**Comment by chobits** [comment]
has been fixed since v0.0.4

**Comment by wskyhawk** [comment]
你好！你的邮件已经收到，我会尽快给你回复，谢谢！
